pkg/providers/imagefamily/customscriptsbootstrap/provisionclientbootstrap_test.go

```diff
@@ -737,3 +820,147 @@ func TestArtifactStreamingEnablement(t *testing.T) {
+			ossku:              customscriptsbootstrap.ImageFamilyOSSKUAzureLinux2,
+			kubernetesVersion:  "1.31.0",
+			imageDistro:        "aks-azurelinux-v2-gen2",
+			fipsMode:           &v1beta1.FIPSModeFIPS,
+			expectedEnableFIPS: true,
+			description:        "FIPS should be enabled for AzureLinux2 with FIPSMode FIPS",
+		},
+		{
+			name:               "FIPSMode nil AzureLinux2 - EnableFIPS false",
+			ossku:              customscriptsbootstrap.ImageFamilyOSSKUAzureLinux2,
+			kubernetesVersion:  "1.31.0",
+			imageDistro:        "aks-azurelinux-v2-gen2",
+			fipsMode:           nil,
+			expectedEnableFIPS: false,
+			description:        "FIPS should be disabled for AzureLinux2 with FIPSMode nil",
+		},
+		{
+			name:               "FIPSMode Disabled AzureLinux2 - EnableFIPS false",
+			ossku:              customscriptsbootstrap.ImageFamilyOSSKUAzureLinux2,
+			kubernetesVersion:  "1.31.0",
+			imageDistro:        "aks-azurelinux-v2-gen2",
+			fipsMode:           &v1beta1.FIPSModeDisabled,
+			expectedEnableFIPS: false,
+			description:        "FIPS should be disabled for AzureLinux2 with FIPSMode Disabled",
+		},
+		{
+			name:               "FIPSMode FIPS AzureLinux3 - EnableFIPS true",
+			ossku:              customscriptsbootstrap.ImageFamilyOSSKUAzureLinux3,
+			kubernetesVersion:  "1.32.0",
+			imageDistro:        "aks-azurelinux-v3-gen2",
+			fipsMode:           &v1beta1.FIPSModeFIPS,
+			expectedEnableFIPS: true,
+			description:        "FIPS should be enabled for AzureLinux3 with FIPSMode FIPS",
+		},
+		{
+			name:               "FIPSMode nil AzureLinux3 - EnableFIPS false",
+			ossku:              customscriptsbootstrap.ImageFamilyOSSKUAzureLinux3,
+			kubernetesVersion:  "1.32.0",
+			imageDistro:        "aks-azurelinux-v3-gen2",
+			fipsMode:           nil,
+			expectedEnableFIPS: false,
+			description:        "FIPS should be disabled for AzureLinux3 with FIPSMode nil",
+		},
+		{
+			name:               "FIPSMode Disabled AzureLinux3 - EnableFIPS false",
+			ossku:              customscriptsbootstrap.ImageFamilyOSSKUAzureLinux3,
+			kubernetesVersion:  "1.32.0",
+			imageDistro:        "aks-azurelinux-v3-gen2",
+			fipsMode:           &v1beta1.FIPSModeDisabled,
+			expectedEnableFIPS: false,
+			description:        "FIPS should be disabled for AzureLinux3 with FIPSMode Disabled",
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			// Create a copy of the base bootstrapper and modify for this test
+			bootstrapper := *baseBootstrapper
+			bootstrapper.OSSKU = tt.ossku
+			bootstrapper.KubernetesVersion = tt.kubernetesVersion
+			bootstrapper.ImageDistro = tt.imageDistro
+			bootstrapper.FIPSMode = tt.fipsMode
+
+			// Setup context with options
+			ctx := options.ToContext(context.Background(), &options.Options{
+				VMMemoryOverheadPercent: 0.075,
+				KubeletIdentityClientID: "test-kubelet-client-id",
+			})
+
+			// Call ConstructProvisionValues
```

Comment:
minor: Comment is not particularly useful.

Comments should (ideally) add something over/above what the code already does (_generally_ say why, not what, unless the "what" is complicated/hard to understand).

In this case, you may not need a comment here.